app/src/main/java/memphis/myapplication/IntroActivity.java

```diff
@@ -1,25 +1,62 @@
 package memphis.myapplication;
 
+import android.content.Context;
 import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
+import android.widget.Toast;
 
 public class IntroActivity extends AppCompatActivity {
 
+    final String nfdAppPackageName = "net.named_data.nfd";
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        Session session = new Session(getApplicationContext());
-        // session checks sharedPreferences where we store our login boolean variable
-        // if we're not logged in, start LoginActivity
-        if (!session.getLoginStatus()) {
-            Intent intent = new Intent(this, LoginActivity.class);
-            startActivity(intent);
+        Context context = getApplicationContext();
+        Session session = new Session(context);
+
+        // Check if NDN Forwarding Daemon is installed.
+        // If installed -> Continue Regular Onboarding
+        // If not installed -> Show a Message and request User to install NDN Forwarding Daemon.
+        PackageManager pm = context.getPackageManager();
+        if(isPackageInstalled(nfdAppPackageName,pm))
+        {
+            // session checks sharedPreferences where we store our login boolean variable
+            // if we're not logged in, start LoginActivity
+            if (!session.getLoginStatus()) {
+                Intent intent = new Intent(this, LoginActivity.class);
+                startActivity(intent);
+            }
+            // we are logged in; take us to MainActivity
+            else {
+                Intent intent = new Intent(this, MainActivity.class);
+                startActivity(intent);
+            }
+        }else{
+
+            Toast.makeText(context,"Please consider Installing NFD, this is " +
```

Comment:
Maybe just "Please install NFD which is required for npChat to work. Start the NFD from within the NFD app and then reopen this app." Also space after comma: context, "..."